File: lingting-spring-elasticsearch/src/test/kotlin/live/lingting/spring/elasticsearch/EntityServiceImpl.java
package live.lingting.spring.elasticsearch;

import org.springframework.stereotype.Component;

/**
 * @author lingting 2024-03-08 18:08
 */
@Component
public class EntityServiceImpl extends AbstractElasticsearchServiceImpl<SpringElasticsearchTest.Entity> {

	@Override
	public String documentId(SpringElasticsearchTest.Entity entity) {
		return entity.getId();
	}

}

File: lingting-spring-elasticsearch/src/test/kotlin/live/lingting/spring/elasticsearch/EntityServiceImpl.kt
package live.lingting.spring.elasticsearch

import org.springframework.stereotype.Component

/**
 * @author lingting 2024-03-08 18:08
 */
@Component
class EntityServiceImpl : AbstractElasticsearchServiceImpl<SpringElasticsearchTest.Entity>() {
    override fun documentId(entity: SpringElasticsearchTest.Entity): String {
        return entity.id
    }
}
